### DISCUSSION TITLE
[v4] Separate prefixes for utilities and theme variables

### DISCUSSION THREAD

**Discussion by andrew-pledge-io** [tailwindlabs/tailwindcss#15093] [Ideas]
Hi, v4 looks great, very excited for some of the changes. In particular, it's great to see built in support for CSS variables.

Apologies if this is already possible (it didn't seem so based on the [prefix section in the beta article](https://tailwindcss.com/docs/v4-beta#using-a-prefix)), but it would be great to be able to declare a separate prefix for the utilities and the generated CSS variables.

Today we use the [tailwindcss-variables](https://github.com/mertasan/tailwindcss-variables) plugin and we have configured a prefix in that plugin that prefixes the generated CSS variables.

We use a prefix on the CSS variables to:
- Avoid conflicts with other code declaring CSS variables (particularly on `:root`)
- Make it easy to discover all theme variables from JavaScript (filtering by prefix)

However we **don't** use a prefix on the Tailwind utility classes as these reasons don't apply there and we prefer the less verbose syntax.

It would be great to make this configuration possible in v4 since I think it might be commonly desired, but also if others are using a similar setup it would help ease the migration.

**Comment by john589dane** [comment]
Hello,
v4 looks fantastic, especially with the built-in support for CSS variables. However, it would be great if we could declare separate prefixes for utilities and generated CSS variables. Using a prefix for CSS variables helps avoid conflicts and makes theme variables easier to discover in JavaScript, while keeping utility classes less verbose. This feature would be a valuable addition and ease migration for those with similar setups. [tiktok downloader](https://www-snaptik.com)

Thanks for considering this suggestion!
